cdap-security/src/main/java/io/cdap/cdap/security/tools/AccessTokenGeneratorService.java

```diff
@@ -0,0 +1,216 @@
+import java.util.Base64;
+import java.util.Collections;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.QueryParam;
+
+/**
+ * A service for managing the lifecycle of an http service that serves access tokens.
+ */
+public class AccessTokenGeneratorService extends AbstractIdleService {
+  static final String ADDRESS_DEFAULT = "127.0.0.1";
+  static final int PORT_DEFAULT = 11030;
+  // HttpHandler has only one caller, and is accessed infrequently so one worker thread is sufficient
+  static final int WORKER_POOL_SIZE_DEFAULT = 1;
+  static final int EXEC_POOL_SIZE_DEFAULT = 0;
+  private final NettyHttpService httpService;
+  private final TokenGeneratorHttpHandler handler;
+
+  AccessTokenGeneratorService(String address, int port, int workerThreadPoolSize, int execThreadPoolSize) {
+    CConfiguration cConf = CConfiguration.create();
+    Injector injector = Guice.createInjector(
+      new ConfigModule(),
+      new IOModule(),
+      new AbstractModule() {
+        @Override
+        protected void configure() {
+          // We don't do anything with metrics, so just bind to the no-op implementation
+          bind(MetricsCollectionService.class).to(NoOpMetricsCollectionService.class);
+        }
+      },
+      CoreSecurityRuntimeModule.getDistributedModule(cConf));
+    CommonNettyHttpServiceFactory factory = injector.getInstance(CommonNettyHttpServiceFactory.class);
+    handler = injector.getInstance(TokenGeneratorHttpHandler.class);
+    NettyHttpService.Builder builder = factory.builder("accesstoken.generator")
+      .setHost(address)
+      .setPort(port)
+      .setExecThreadPoolSize(execThreadPoolSize)
+      .setWorkerThreadPoolSize(workerThreadPoolSize)
+      .setExceptionHandler(new HttpExceptionHandler())
+      .setHttpHandlers(handler);
+    if (cConf.getBoolean(Constants.Security.SSL.INTERNAL_ENABLED)) {
+      SConfiguration sConf = injector.getInstance(SConfiguration.class);
+      new HttpsEnabler().configureKeyStore(cConf, sConf).enable(builder);
+    }
+    httpService = builder.build();
+  }
+
+  @VisibleForTesting
+  InetSocketAddress getBindAddress() {
+    return httpService.getBindAddress();
+  }
+
+  @VisibleForTesting
+  TokenManager getTokenManager() {
+    return handler.tokenManager;
+  }
+
+  @VisibleForTesting
+  Codec<AccessToken> getTokenCodec() {
+    return handler.tokenCodec;
+  }
+
+  @Override
+  protected void startUp() throws Exception {
+    httpService.start();
+  }
+
+  @Override
+  protected void shutDown() throws Exception {
+    httpService.stop();
+  }
+
+  @Path(Constants.Gateway.INTERNAL_API_VERSION_3 + "/accesstoken")
+  public static class TokenGeneratorHttpHandler extends AbstractHttpHandler {
+    private final CConfiguration cConf;
+    final TokenManager tokenManager;
+    final Codec<AccessToken> tokenCodec;
+
+    @Inject
+    public TokenGeneratorHttpHandler(CConfiguration cConf, TokenManager tokenManager,
```

Comment:
remove `public` for injected constructor

Reply:
fixed